src/Swarm/TUI/Model.hs

```diff
@@ -678,6 +721,35 @@ populateInventoryList (Just r) = do
 
 -- | Initialize the 'AppState'.
 initAppState :: Maybe Seed -> Maybe String -> Maybe String -> Bool -> ExceptT Text IO AppState
-initAppState seed scenarioName toRun cheatMode = do
-  let showMenu = isNothing scenarioName && isNothing toRun && isNothing seed
-  AppState <$> initGameState seed scenarioName toRun <*> initUIState showMenu cheatMode
+initAppState userSeed scenarioName toRun cheatMode = do
+  let skipMenu = isJust scenarioName || isJust toRun || isJust userSeed
+  gs <- initGameState
+  ui <- initUIState (not skipMenu) cheatMode
+  case skipMenu of
+    False -> return $ AppState gs ui
+    True -> do
+      scenario <- loadScenario (fromMaybe "00-classic" scenarioName) (gs ^. entityMap)
+      liftIO $ scenarioToAppState scenario userSeed toRun $ AppState gs ui
+
+-- XXX do we need to keep an old entity map around???
```

Comment:
I do not think so, we can always reload it if we wanted to say compare it before saving the game (or just save the whole map). Do you have some use cases in mind?

Reply:
I'm worried that when we load a scenario with custom entities, they get *permanently* added to the entity map...

Reply:
Well, that's unrelated to this PR though, and it would have already been an issue before.  I'll think about it later.